GLMakie/assets/shader/mesh.frag

```diff
@@ -29,17 +29,21 @@ vec4 get_color(Nothing color, vec2 uv, vec2 color_norm, sampler1D color_map, Not
     return o_color;
 }
 
-vec4 get_color(sampler2D color, vec2 uv, vec2 color_norm, sampler1D color_map, Nothing matcap){
-    return o_color;
-}
-
 vec4 get_color(sampler2D color, vec2 uv, Nothing color_norm, Nothing color_map, Nothing matcap){
     return texture(color, uv);
 }
 
-vec4 color_lookup(float intensity, sampler1D color, vec2 norm);
+float _normalize(float val, float from, float to){return (val-from) / (to - from);}
+vec4 color_lookup(float intensity, sampler1D color_ramp, vec2 norm){
+    return texture(color_ramp, _normalize(intensity, norm.x, norm.y));
+}
+
+uniform vec4 highclip;
+uniform vec4 lowclip;
+uniform vec4 nan_color;
+
 vec4 get_color(sampler2D intensity, vec2 uv, vec2 color_norm, sampler1D color_map, Nothing matcap){
-    const float i = texture(intensity, uv).x;
+    float i = texture(intensity, uv).x;
     vec4 color = color_lookup(i, color_map, color_norm);
     if (isnan(i)) {
```

Comment:
isnan shall be checked before color_lookup; unnecessary call to color lookup for nan values

Reply:
Yeah makes sense... I copy and pasted this from an older shader, and it feels like I had a reason for this. I'll try it out tomorrow